main.c

```diff
@@ -973,15 +1014,29 @@ int main(int argc, char *argv[]) {
 
 	struct wl_registry *registry = wl_display_get_registry(state.display);
 	wl_registry_add_listener(registry, &registry_listener, NULL);
+	wl_display_dispatch(state.display);
 	wl_display_roundtrip(state.display);
 
+	struct seat **ptr;
+    wl_array_for_each(ptr, &state.seats) {
+        struct seat *seat_p = *ptr;
+        if (state.seat_name == NULL || strcmp(seat_p->name, state.seat_name) == 0) {
+			seat = seat_p->proxy;
+        }
+    }
+
 	if (idle_manager == NULL) {
 		swayidle_log(LOG_ERROR, "Display doesn't support idle protocol");
 		swayidle_finish();
 		return -4;
 	}
 	if (seat == NULL) {
-		swayidle_log(LOG_ERROR, "Seat error");
+		if (state.seat_name != NULL) {
+			swayidle_log(LOG_ERROR, "Seat %s not found", state.seat_name);
+		}
+		else {
+			swayidle_log(LOG_ERROR, "Seat error");
```

Comment:
"No seat found" is maybe a better error message? (I see this error message was there before…)